docs/setup/independent/troubleshooting-kubeadm.md

```diff
@@ -0,0 +1,143 @@
+
+This is expected and part of the design. kubeadm is network provider-agnostic, so the admin
+should [install the pod network solution](/docs/concepts/cluster-administration/addons/)
+of choice. You have to install a Pod Network
+before `kube-dns` may deployed fully. Hence the `Pending` state before the network is set up.
+
+#### `HostPort` services do not work
+
+The `HostPort` and `HostIP` functionality is available depending on your Pod Network
+provider. Please contact the author of the Pod Network solution to find out whether
+`HostPort` and `HostIP` functionality are available.
+
+If not, you may still use the [NodePort feature of
+services](/docs/concepts/services-networking/service/#type-nodeport) or use `HostNetwork=true`.
+
+#### Pods are not accessible via their Service IP
+
+Many network add-ons do not yet enable [hairpin mode](https://kubernetes.io/docs/tasks/debug-application-cluster/debug-service/#a-pod-cannot-reach-itself-via-service-ip)
+which allows pods to access themselves via their Service IP if they don't know about their podIP. This is an issue
+related to [CNI](https://github.com/containernetworking/cni/issues/476). Please contact the providers of the network
+add-on providers to get timely information about whether they support hairpin mode.
+
+If you are using VirtualBox (directly or via Vagrant), you will need to
+ensure that `hostname -i` returns a routable IP address (i.e. one on the
+second network interface, not the first one). By default, it doesn't do this
+and kubelet ends-up using first non-loopback network interface, which is
+usually NATed. Workaround: Modify `/etc/hosts`, take a look at this
+`Vagrantfile`[ubuntu-vagrantfile](https://github.com/errordeveloper/k8s-playground/blob/22dd39dfc06111235620e6c4404a96ae146f26fd/Vagrantfile#L11) for how this can be achieved.
+
+#### TLS certificate errors
+
+The following error indicates a possible certificate mismatch.
+
+```
+# kubectl get po
+Unable to connect to the server: x509: certificate signed by unknown authority (possibly because of "crypto/rsa: verification error" while trying to verify candidate authority certificate "kubernetes")
+```
+
+Verify that the `$HOME/.kube/config` file contains a valid certificate, and regenerate a certificate if necessary.
+Another workaround is to overwrite the default `kubeconfig` for the "admin" user:
+
+```
+mv  $HOME/.kube $HOME/.kube.bak
+mkdir -p $HOME/.kube
+sudo cp -i /etc/kubernetes/admin.conf $HOME/.kube/config
+sudo chown $(id -u):$(id -g) $HOME/.kube/config
+```
+
+#### Errors on CentOS when setting up masters
+
+If you are using CentOS and encounter difficulty while setting up the master node，
+verify that your Docker cgroup driver matches the kubelet config:
+
+```bash
+docker info |grep -i cgroup
+cat /etc/systemd/system/kubelet.service.d/10-kubeadm.conf
+```
+
+If the Docker cgroup driver and the kubelet config don't match, change the kubelet config to match the Docker cgroup driver.
+
+Update
```

Comment:
Can you instead give the sed command to run so it can easily be done non-interactively?